network/wsNetwork.go

```diff
@@ -1677,6 +1725,17 @@ func (wn *WebsocketNetwork) OnNetworkAdvance() {
 	wn.lastNetworkAdvanceMu.Lock()
 	defer wn.lastNetworkAdvanceMu.Unlock()
 	wn.lastNetworkAdvance = time.Now().UTC()
+	if wn.nodeInfo != nil && !wn.relayMessages && !wn.config.ForceFetchTransactions {
+		// if we're not a relay, and not participating, we don't need txn pool
+		wantTXGossip := wn.nodeInfo.IsParticipating()
```

Comment:
inside IsParticipating is a call to Ledger.Latest() (which takes a lock on blockQueue.mu.Lock()) and accountManager.HasLiveKeys() (which takes a lock on manager.mu.Lock()).

Before this PR, OnNetworkAdvance used to be very fast — just updating a timestamp. Now it could get stuck for a long time, I know that accountManager.DeleteOldKeys() (which also takes the manager.mu.Lock()) could take seconds in some cases.

Is it possible that OnNetworkAdvance could take too long — agreement synchronously waits for OnNetworkAdvance to finish at the end of every round, when calling the node's EnsureBlock/EnsureValidatedBlock methods. This means the performance of OnNetworkAdvance will delay the start of agreement beginning the next round.